plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -89,7 +89,42 @@ func LeastRequestedPriority(pod *api.Pod, podLister algorithm.PodLister, minionL
 
 	list := algorithm.HostPriorityList{}
 	for _, node := range nodes.Items {
-		list = append(list, calculateOccupancy(pod, node, podsToMachines[node.Name]))
+		list = append(list, calculateResourceOccupancy(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func min(l, r int64) (m int64) {
+	m = r
+	if l < r {
+		m = l
+	}
+	return m
+}
+
+// See comment for DumbSpreadingPriority()
+const dumbSpreadingDenominator int64 = 10
+
+// DumbSpreadingPriority is a priority function that favors nodes with fewer pods.
+// It works like LeastRequestedPeriority but instead of using 10 * percentage of machine free by resource,
+// it uses 10 * percentage of machine free by pod, with "percentage of machine free by pod" claculated as
+// (dumbSpreadingDenominator - number of pods already on the node + 1) / dumbSpreadingDenominator.
+// dumbSpreadingDenominator serves like the machine capacity in LeasRequestedPriority but is chosen
+// so that we equate one pod with a reasonable amount of resources when we combine all the scores together.
```

Comment:
I don't think I agree with everything you said, but you said many things, so there are many things to agree or disagree with. :)

(1) I think one thing you said is that I should be using LeastRequestedPriority ("It isn't clear why this is better than leastrequester"). If so, I don't understand -- I basically *am* using LeastRequestedPriority, just splitting it into a separate function and using pods instead of resource values (same as your PR did). I might be misinterpreting this comment though.

(2) You said "tell this algorithm to outweigh everything else in the no resource specified case and keep completely quiet otherwise." I don't agree with that. If one machine has one pod that we know takes up 99% of the machine (by limit) and another machine has one pod with 1% limit, I don't think we want to evenly distribute the next N arriving 0-limit pods between those two machines. Conversely, as @bgrant0607 mentioned earlier, we probably want to consider dumb spreading even when scheduling non-0-limit pods. So I don't think we ever want to "turn off" either the resource-based priority or the pod-spreading-based priority. Both resources and spreading should always contribute something when scheduling any kind of pod.

(3) You pointed out that the approach I'm using "maxes out" at contributing 50% towards the final score when there are 10 pods. So even if a node has 1000 pods that each request 0 or some tiny limit, we might continue to schedule more pods onto it. This is true but I think this only happens in extreme corner cases. For example in a case like

node1: total limit 95% and 1 pod
node2: total limit 20% and 10 pods
we will still put the next pod on node1. So I think that within reasonable ranges the simple weighting I'm using works OK. I'm really reluctant to go to a dynamic weighting mechanism so close to the code deadline. As long as we handle the case where the user starts out with the standard cluster addon pods, and then schedules a bunch of 0-limit pods, those pods should be reasonably evenly spread, then I think it's good enough for now. I do need to verify that's the case (I need to see what limits we're giving the addon pods and calculate) but I suspect it is.